jerry-core/ecma/operations/ecma-objects-general.c

```diff
@@ -201,6 +223,45 @@ ecma_op_general_object_default_value (ecma_object_t *obj_p, /**< the object */
   JERRY_ASSERT (obj_p != NULL
                 && !ecma_is_lexical_environment (obj_p));
 
+#if ENABLED (JERRY_ES2015)
+  ecma_value_t exotic_to_prim = ecma_op_get_method_by_symbol_id (ecma_make_object_value (obj_p),
+                                                                 LIT_MAGIC_STRING_TO_PRIMITIVE);
+
+  if (ECMA_IS_VALUE_ERROR (exotic_to_prim))
+  {
+    return exotic_to_prim;
+  }
+
+  if (!ecma_is_value_undefined (exotic_to_prim))
+  {
+    ecma_object_t *call_func_p = ecma_get_object_from_value (exotic_to_prim);
+    ecma_value_t argument = ecma_make_magic_string_value (hints[hint]);
+
+    ecma_value_t result = ecma_op_function_call (call_func_p,
+                                                 ecma_make_object_value (obj_p),
```

Comment:
The `obj_p` is converted to object value two times. Would it be possible to create the object value only once?
Also do need to wrap it into an objet value? It would be great if (even in a follow up pr) to reduce the number of conversions around this call (or changing the `ecma_op_get_method_by_symbol_id` signature).

Reply:
Is there any test for this `toPrimitive` invocation?